lib/system/alloc.nim

```diff
@@ -908,7 +912,7 @@ proc rawAlloc(a: var MemRegion, requestedSize: int): pointer =
       if c.free >= size:
         # Because removals from `a.freeSmallChunks[s]` only happen in the other alloc branch and during dealloc,
         #  we must not add it to the list if it cannot be used the next time a pointer of `size` bytes is needed.
-        listAdd(a.freeSmallChunks[s], c)
+        a.freeSmallChunks[s] = c
```

Comment:
This makes me nervous. What is the justification for it?

Reply:
There is always only a single active chunk which receives cells when they get freed, so there is no reason for chunks to keep acting as a list in the way they did before.

Reply:
I was thinking of using the list behavior for the remaining threading issue to easily find chunks that must stay alive, but not sure if I can figure out how to do that yet.